dap_aria_mapping/pipeline/make_topic_name_assignments.py

```diff
@@ -0,0 +1,432 @@
+                }
+            ),
+        }
+
+    for taxlabel, taxonomy in taxonomies:
+        logger.info("Starting taxonomy {}".format(taxlabel))
+        for level in [int(x) for x in args.levels]:
+            logger.info("Starting level {}".format(str(level)))
+            logger.info("Building dictionary - entity to journals")
+            if any(["entity" in args.name_type, "journal" in args.name_type]):
+                clust_counts = get_level_entity_counts(
+                    journal_entities, taxonomy, level=level
+                )
+
+            if "entity" in args.name_type:
+                logger.info("Building dictionary - entity to names")
+                names = get_cluster_entity_names(
+                    clust_counts, num_entities=args.n_top, show_count=args.show_count
+                )
+                if args.save:
+                    logger.info("Saving dictionary as pickle")
+                    save_names(
+                        taxonomy=taxlabel,
+                        name_type="entity",
+                        level=level,
+                        n_top=args.n_top,
+                        names=names,
+                    )
+
+            if "journal" in args.name_type:
+                logger.info("Building dictionary - journal to names")
+                cluster_journal_ranks = get_cluster_journal_ranks(
+                    clust_counts, journal_entities, output="relative"
+                )
+                names = get_cluster_journal_names(
+                    cluster_journal_ranks, num_names=args.n_top
+                )
+                if args.save:
+                    logger.info("Saving dictionary as pickle")
+                    save_names(
+                        taxonomy=taxlabel,
+                        name_type="journal",
+                        level=level,
+                        n_top=args.n_top,
+                        names=names,
+                    )
+
+            if "chatgpt" in args.name_type:
+                entity_names = get_topic_names(
+                    taxonomy_class=taxlabel,
+                    name_type="entity",
+                    level=level,
+                    n_top=args.n_top,
+                )
+
+                files_with_name = [
```

Comment:
I think we have a function that does this already in nesta_ds_utils: `nesta_ds_utils.loading_saving.S3.get_bucket_filenames_s3`